src/index.ts

```diff
@@ -299,7 +299,7 @@ function buildFilter<T>(filters: Filter<T> = {}, aliases: Alias[] = [], propPref
                   result.push(`${op}(${propName},${handleValue(value[op], aliases)})`);
                 } else {
                   // Nested property
-                  const filter = buildFilterCore(value, aliases, propName);
+                  const filter = buildFilterCore({ [op]: value[op] }, aliases, propName);
```

Comment:
This is the heart of the change.  We don't want to build the filter against the entire object; we just want to use the portion that's applicable to this iteration of the for-loop.